### ISSUE TITLE
MiKo_1040/1041/1070 shall not report in case a type inherits from an interface that only indirectly inherits from IEnumerable

### ISSUE THREAD

**Issue by RalfKoban** [RalfKoban/MiKo-Analyzers#1893]
In case we have an interface that indirectly inherits from `IEnumerable`, we should ignore such variables and parameters (and not report any issue in case it is not in plural form).

Example:
```csharp
public interface IMyStuff : IWhatever
{}

public interface IWhatever : IEnumerable
{}

public class TestMe
{
    public void DoSomething(IMyStuff myStuff) // this should not trigger
    {
       var myOtherStuff = myStuff; // this should not trigger
    }
    private IMyStuff myRemainingStuff; // this should not trigger
}
```
